aten/src/ATen/native/cpu/PointwiseOpsKernel.cpp

```diff
@@ -46,28 +46,31 @@ static void addcdiv_cpu_kernel(TensorIterator& iter, Scalar value) {
   });
 }
 
-static void smooth_l1_backward_cpu_kernel(TensorIterator& iter, Scalar norm) {
+static void smooth_l1_backward_cpu_kernel(TensorIterator& iter, Scalar norm, Scalar beta) {
   ScalarType dtype = iter.dtype(0);
   AT_DISPATCH_ALL_TYPES(dtype, "smooth_l1_backward_cpu_out", [&] {
     auto norm_val = norm.to<scalar_t>();
     auto norm_val_vec = Vec256<scalar_t>(norm_val);
+    auto beta_val = beta.to<scalar_t>();
+    auto beta_val_vec = Vec256<scalar_t>(beta_val);
     const auto neg_1_vec = Vec256<scalar_t>(-1);
     const auto pos_1_vec = Vec256<scalar_t>(1);
     cpu_kernel_vec(iter,
       [=](scalar_t input, scalar_t target, scalar_t grad_output) -> scalar_t {
         const auto x = input - target;
-        if (x < -1.)
+        if (x < -beta_val)
           return -norm_val * grad_output;
-        else if (x > 1.)
+        else if (x > beta_val)
           return norm_val * grad_output;
         else
-          return norm_val * x * grad_output;
+          return norm_val * x * grad_output / beta_val;
       },
-      [norm_val_vec, neg_1_vec, pos_1_vec](
+      [norm_val_vec, beta_val_vec, neg_1_vec, pos_1_vec](
          Vec256<scalar_t> input, Vec256<scalar_t> target, Vec256<scalar_t> grad_output) -> Vec256<scalar_t> {
         auto x = input - target;
+        x /= beta_val_vec;
```

Comment:
performance question:

I kept the clamping logic instead of using conditionals just to minimize the amount of change that I introduce. But that comes with the downside of always having to divide by beta_val_vec. Versus the conditional code would look something like (only have to do the division in the last case):
`if (x > beta_val_vec) return pos_1_vec else if (x < -beta_val_vec) return neg_1_vec else return norm_val_vec * x * grad_output / beta_val_vec`

How carefully should we be looking at performance for a change like this?